images/metering-ansible-operator/roles/meteringconfig/tasks/configure_tls.yml

```diff
@@ -0,0 +1,60 @@
+---
+
+- name: Check if we need to create a Presto TLS secret using the user-populated fields
+  block:
+  - name: Validate that the user provided a non-empty TLS certificate value
+    fail:
+      msg: "presto.spec.presto.config.tls.certificate cannot be empty if createSecret: true and secretName != ''"
+    when: meteringconfig_spec.presto.spec.presto.config.tls.certificate == ""
+
+  - name: Validate that the user provided a non-empty TLS CA certificate value
+    fail:
+      msg: "presto.spec.presto.config.tls.caCertificate cannot be empty if createSecret: true and secretName != ''"
+    when: meteringconfig_spec.presto.spec.presto.config.tls.caCertificate == ""
+
+  - name: Validate that the user provided a non-empty TLS key value
+    fail:
+      msg: "presto.spec.presto.config.tls.key cannot be empty if createSecret: true and secretName != ''"
+    when: meteringconfig_spec.presto.spec.presto.config.tls.key == ""
+  when: meteringconfig_template_presto_tls_secret
+
+- name: Check if we need to create a Presto TLS secret using the user-populated fields
```

Comment:
The name here doesn't match the logic. It's just doing validation on the Presto Auth TLS configuration.